src/Compiler/TypedTree/TypedTree.fs

```diff
@@ -867,10 +867,10 @@ type Entity =
     member x.IsModuleOrNamespace = x.entity_flags.IsModuleOrNamespace
 
     /// Indicates if the entity is a namespace
-    member x.IsNamespace = x.IsModuleOrNamespace && (match x.ModuleOrNamespaceType.ModuleOrNamespaceKind with Namespace -> true | _ -> false)
```

Comment:
Adding an `member x.IsImplicitNamespace` might be useful here at various other points in the code?